src/components/SignIn.js

```diff
@@ -55,6 +129,16 @@ export default function SignIn() {
         <Typography component="h1" variant="h5">
           Sign in
         </Typography>
+        {/* display snackbar if any error happened during user login */}
+        {loginError && (
+          <Snackbar
+            open={open}
+            autoHideDuration={6000}
+            onClose={handleClose}
+            message={errorMessage}
+            action={action}
+          />
+        )}
```

Comment:
The Snackbar component handles by itself internally whether to be displayed or not (based on things like the `autoHideDuration` that you give it. So we should not need to do further conditional rendering logic, and in fact it may lead to sneaky bugs:

```suggestion
        {/* display snackbar if any error happened during user login */}
          <Snackbar
            open={open}
            autoHideDuration={6000}
            onClose={handleClose}
            message={errorMessage}
            action={action}
          />
```

We just need to make sure we only set `open` to `true` when we want to display the Snackbar (eg: in the `handleLoginError` function)

Reply:
You are right, I missed that point